### PR TITLE
HWS Series

### PR DESCRIPTION
**Pull request by hamdanigbaria**

### CODE REVIEW

File: hws/mlx_hw_steering_parser.py [hunk 1]
```diff
@@ -195,7 +207,9 @@ def parse_args():
                         help="Trigger DPDK app <PID>.")
     parser.add_argument("-port", dest="dpdk_port", type=int, default=0,
                         help="Trigger DPDK app <PORT> newer dpdk supports -1 for all ports (must provide PID with -pid).")
-    parser.add_argument('-h', '--help', action='help', default=argparse.SUPPRESS,
+    parser.add_argument("-extra_hw_res", type=str, default="", dest="extra_hw_res", metavar="[pat, arg]",
+                        help = "Provide extra HW resources to be dumped. for example: -extra_hw_res pat,arg")
```
Comment (kliteyn) [inline hws/mlx_hw_steering_parser.py:211]: "Provide extra HW resources to be dumped"
Perhaps "require"? "request"? We're not providing - we're just asking to dump more resources

File: hws/src/dr_common.py [hunk 2]
```diff
@@ -133,3 +141,233 @@ def hex_to_bin_str(_n, _len):
...
+    0x33: "IN2_IP_START",
+    0x34: "IN2_ESP",
+    0x35: "IN2_TCP_UDP_START",
+    0x36: "IN_TUNNEL_HEADER_START",
+    0x37: "IN2_L2_END",
+    0x38: "IN2_FIRST_CFG_ETHER-TYPE_END",
+    0x39: "IN2_SECOND_CFG_ETHER-TYPE_END",
+    0x3A: "IN2_MPLS_END",
+    0x3B: "IN2_IP_END",
+}
```
Comment (kliteyn) [inline hws/src/dr_common.py:373]: add new line

File: hws/src/dr_context.py [hunk 3]
```diff
@@ -63,7 +63,7 @@ def tree_print(self, verbosity, tabs):
         return _str
 
     def fix_data(self):
-        self.data["hws_support"] = "True" if self.data["hws_support"] is "1" else "False"
+        self.data["hws_support"] = "True" if self.data["hws_support"] == "1" else "False"
```
Comment (kliteyn) [inline hws/src/dr_context.py:66]: not in this patch

File: hws/src/dr_matcher.py [hunk 4]
```diff
@@ -215,20 +215,22 @@ def __init__(self, data):
...
+
     def dump_str(self, verbosity):
-        if verbosity > 0:
-            return dump_obj_str(["mlx5dr_debug_res_type",
-                                 "priority", "mode", "sz_row_log",
-                                 "sz_col_log", "flow_src", "insertion",
-                                 "distribution"], self.data)
+        _keys = ["mlx5dr_debug_res_type", "priority", "log_sz"]
+        if verbosity > 1:
+            _keys.extend(["mode", "flow_src"])
```
Comment (kliteyn) [inline hws/src/dr_matcher.py:222]: So the default verbosity is 0, right?
What happens with the verbosity of 1?

File: hws/src/dr_ste.py [hunk 5]
```diff
@@ -81,7 +81,7 @@ def fields_handler(_fields, verbosity=0, show_field_val=False):
 
 def ste_hit_addr_calc(next_table_base_63_48, next_table_base_39_32, next_table_base_31_5):
     hit_addr = next_table_base_39_32 << 32
-    hit_addr |= next_table_base_31_5
+    hit_addr |= (next_table_base_31_5 << 5)
```
Comment (kliteyn) [inline hws/src/dr_ste.py:84]: so we had this issue all the time... :)

**Review by kliteyn — CHANGES_REQUESTED** [review]
There are couple comments and some things we need to discuss

File: hws/src/dr_ste.py (hunk 5 above)
Comment (hamdanigbaria) [reply hws/src/dr_ste.py:84]: will fix the miss, but it is not in use

Comment (hamdanigbaria) [reply hws/src/dr_ste.py:84]: fix in other patch on top of series in a fixes patch

**Pull request closed without merging**